### PR TITLE
fix(walletFactory): move upgrading check before baggage is populated

### LINKED ISSUE #8321: walletFactory fails to install bridge handler on first incarnation
## Describe the bug

This `upgrading` check is supposed to distinguish the 1st incarnation from others, but it happens after `provideDurableMapStore(baggage, 'walletsByAddress')` so it's always `true` and in the 1st incarnation, the `setHandler` call fails:

https://github.com/Agoric/agoric-sdk/blob/d35503027f8de03dfedf7bc3a5adce00d0985435/packages/smart-wallet/src/walletFactory.js#L295-L300

## To Reproduce
1. in `packages/cosmic-swingset`: `make scenario2-setup BASE_PORT=8000 NUM_SOLOS=0;`
2. make scenario2-run-chain
3. choose an address; use `make fund-acct` to provide BLD and `make provision-acct` to provision a smart wallet
4. sign and broadcast an offer from the address
5. see `Error#5: No inbound handler for wallet`

## Expected behavior

offers handled as usual

## Platform Environment
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)
d35503027

## Additional context

https://github.com/Kryha/KREAd/tree/develop/agoric#kread-deploy-steps beb28dc

cc @carlos-kryha

[Aug 24 discussion of `upgrading` check](https://github.com/Agoric/agoric-sdk/pull/8147#discussion_r1304602549)

cc @turadg 

## Logs / Stacktraces

When an offer is submitted:

```
2023-09-11T14:09:13.926Z SwingSet: xsnap: v10: UnhandledPromiseRejectionWarning: (Error#5)
2023-09-11T14:09:13.926Z SwingSet: xsnap: v10: Error#5: No inbound handler for wallet
2023-09-11T14:09:13.926Z SwingSet: xsnap: v10: Error: No inbound handler for (a string)
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:273)
 at fail (/bundled-source/.../node_modules/ses/src/error/assert.js:387)
 at baseAssert (/bundled-source/.../node_modules/ses/src/error/assert.js:407)
 at fromBridge (.../vats/src/bridge.js:62)
```

after re-starting the chain, during bootstrap:

```
2023-09-11T15:36:29.225Z SwingSet: ls: v10: Error#1: Bridge handler not yet initialized for wallet
2023-09-11T15:36:29.225Z SwingSet: ls: v10: Error: Bridge handler not yet initialized for (a string)
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:273)
 at fail (/bundled-source/.../node_modules/ses/src/error/assert.js:387)
 at setHandler (.../vats/src/bridge.js:80)
```

### PR DESCRIPTION
**Pull request by dckc**
closes: #8321
refs: #8311

## Description

The `baggage.has('walletsByAddress')` check has to come before the `provideDurableMapStore(baggage, 'walletsByAddress')` call.

### Testing Considerations

 - [x] IOU a unit test

Meanwhile, @carlos-kryha , if you find a moment to try this fix, please let us know how it goes.

### Security / Scaling / Documentation Considerations

none

### Upgrade Considerations

This fixes the non-upgrade case, i.e. the 1st incarnation case.

### CODE REVIEW

**Comment by carlos-kryha** [comment]
It works, no bridge handler error when starting the chain and I'm able to sign transactions via keplr. Thanks for the fix!

File: packages/smart-wallet/src/walletFactory.js [hunk 1]
```diff
@@ -142,6 +142,7 @@ export const makeAssetRegistry = assetPublisher => {
  * @param {import('@agoric/vat-data').Baggage} baggage
  */
 export const prepare = async (zcf, privateArgs, baggage) => {
+  const upgrading = baggage.has('walletsByAddress');
```
Comment (turadg) [inline packages/smart-wallet/src/walletFactory.js:145]: 👌 

Any ideas for catching the temptation in the future to move such a check?

@Chris-Hibbert WDYT of the zcf providing this sort of info? Perhaps even incarnationNumber?

**Review by turadg — CHANGES_REQUESTED** [review]
Fix looks good. Glad the branch unblocks Kryha.

Requesting changes for code formatting and a regression test. An alternative is to spec and schedule a design that prevents this sort of error.

File: packages/smart-wallet/src/walletFactory.js (hunk 1 above)
Comment (dckc) [reply packages/smart-wallet/src/walletFactory.js:145]: I think our mock bridge is lax about registration. So I expect that if we make it more strict we could reproduce this in a unit test.

**Comment by dckc** [comment]
> Requesting changes for code formatting and ...

oh no! vs-code has gone rogue! format-on-save was my friend but now it's my enemy.
I wonder how to fix this.

File: packages/smart-wallet/test/supports.js [hunk 2]
```diff
@@ -76,9 +76,11 @@ const makeFakeBridgeManager = () =>
           return E(handler).fromBridge(obj);
         },
         initHandler(newHandler) {
+          !handler || Fail`Handler already set`;
           handler = newHandler;
         },
         setHandler(newHandler) {
+          handler || Fail`Handler not set`;
```
Comment (turadg) [inline packages/smart-wallet/test/supports.js]: eslint giving an error. would it work to use the check style on line 70?
```
          if (!handler) {
            Fail`No handler for ${bridgeId}`;
          }
```

Comment (dckc) [reply packages/smart-wallet/test/supports.js]: Oh. I suppose that would have been good for consistency.
I just pushd a fixup with `!!handler || Fail...` though.

**Review by turadg — APPROVED** [review]

**Pull request merged**